Update os-diff doc with package install instructions and correct usages (#452)
openstack-k8s-operators/data-plane-adoption: docs_user/modules/proc_pulling-configuration-from-a-tripleo-deployment.adoc

+
Make sure the ssh command you provide in `ssh_cmd` parameter is correct and with key authentication.

. Enable or disable the services that you want in the `/etc/os-diff/config.yaml` file. Make sure that you have the correct rights to edit the file, for example:
+
----
chown ospng:ospng /etc/os-diff/config.yaml
----

> Example with default {identity_service_first_ref}:
> +

Reviewer: ```suggestion
Example with default {identity_service_first_ref}:

```

+
Make sure the ssh command you provide in `ssh_cmd` parameter is correct and with key authentication.

. Enable or disable the services you want in /etc/os-diff/config.yaml:

> Make sure you have the correct rights to edit the file example:

Reviewer 2: s/example/for example/ ? Alternatively the chown example could also be removed i think.

> . Enable or disable the services you want in /etc/os-diff/config.yaml:
> 
> Make sure you have the correct rights to edit the file, for example:
> 
> ---
> chown ospng:ospng /etc/os-diff/config.yaml
> ---

Reviewer: ```suggestion
. Enable or disable the services that you want in the `/etc/os-diff/config.yaml` file. Make sure that you have the correct rights to edit the file, for example:
+
----
chown ospng:ospng /etc/os-diff/config.yaml
----
```

> +

Reviewer: ```suggestion

```

+
----
services:
  yum_config:
    hosts:
      - undercloud
      - controller_1
      - compute_1
      - compute_2
    service_command: "cat /etc/yum.conf"
    cat_output: true
    path:
      - yum.conf
----

. Pull the configuration:
+

This command will pull all the configuration files that are described in the `/etc/os-diff/config.yaml` file.
Os-diff can update this file automatically according to your running environment with the command `--update` or `--update-only`.
This option sets the podman information into the `config.yaml` for all running containers.
It can be useful later, when all the {rhos_prev_long} services are turned off.

Note that when the `config.yaml` file is populated automatically you must provide the configuration paths manually for each service.

----
# will only update the /etc/os-diff/config.yaml
os-diff pull --update-only
----

----
# will update the /etc/os-diff/config.yaml and pull configuration
os-diff pull --update
----

----
# will update the /etc/os-diff/config.yaml and pull configuration
os-diff pull
----

> +
> The configuration will be pulled and stored by default:
> +

Reviewer: ```suggestion
The configuration will be pulled and stored by default:
```

+
Repeat this step for each {OpenStackShort} service that you want to disable or enable.

> Os-diff can pull configuration or command output from non-containerized servies such as the ovs-external-ids:
> 
> ----
> services:
>   ovs_external_ids:

Reviewer: ```suggestion
. If you are using non-containerized services,  such as the ovs-external-ids, os-diff can pull configuration or command output:
+
----
services:
  ovs_external_ids:
```

    hosts:
      - standalone
    service_command: "ovs-vsctl list Open_vSwitch . | grep external_ids | awk -F ': ' '{ print $2; }'"
    cat_output: true
    path:
      - ovs_external_ids.json
    config_mapping:
      ovn-bridge-mappings: edpm_ovn_bridge_mappings
      ovn-bridge: edpm_ovn_bridge
      ovn-encap-type: edpm_ovn_encap_type
      ovn-match-northd-version: ovn_match_northd_version
      ovn-monitor-all: ovn_monitor_all
      ovn-remote-probe-interval: edpm_ovn_remote_probe_interval
      ovn-ofctrl-wait-before-clear: edpm_ovn_ofctrl_wait_before_clear

> ----
> 
> This service is not an Openstack service executed in a container, so the description and the behavior is different.
> It's important to correctly configure an SSH config file or equivalent for non standard services such as OVS.
> The ovs_external_ids is not a service which runs in a container and the ovs data is stored
> on each host of our cloud: controller_1/controller_2/...

Reviewer: ```suggestion
----
+
This service is not an {rhos_prev_long} service executed in a container, so the description and the behavior is different. It is important to correctly configure an SSH config file or equivalent for non-standard services such as OVS. The `ovs_external_ids` does not run in a container, and the ovs data is stored
on each host of our cloud: controller_1/controller_2/...
```

> The hosts key in the config.yaml will allow os-diff to loop over all hosts specified the output of the command, or the file or data that you need to pull from our deployment in order to compare it later:

Reviewer 2: "... will allow os-diff to loop over all hosts specified the output of the command" -- doesn't make sense i think?

> 
> With the `hosts` key, os-diff will loop on each host provided and execute the command in `service_command` key:
> 
> ----
>     ovs_external_ids:
>         path:
>             - ovs_external_ids.json
>         hosts:
>             - standalone
> ----

Reviewer: ```suggestion
+
With the `hosts` key, os-diff loops on each host and runs the command in the `service_command` key:

----
    ovs_external_ids:
        path:
            - ovs_external_ids.json
        hosts:
            - standalone
----
```

> 
> The service_command is the command which provides the required information.
> It could be a simple cat from a config file. cat_output should be set to true
> if you want os-diff to get the output of the command and store the output in a file specified by the key path
> 
> Then you can provide a mapping between in this case the EDPM CRD, and the ovs-vsctl output with config_mapping
> 
> ----
>         service_command: 'ovs-vsctl list Open_vSwitch . | grep external_ids | awk -F '': '' ''{ print $2; }'''

Reviewer: ```suggestion
+
The `service_command`  provides the required information. It could be a simple cat from a config file. If you want os-diff to get the output of the command and store the output in a file specified by the key path, set `cat_output` to true. Then you can provide a mapping between in this case the EDPM CRD, and the ovs-vsctl output with config_mapping:
+
----
        service_command: 'ovs-vsctl list Open_vSwitch . | grep external_ids | awk -F '': '' ''{ print $2; }'''
```
@matbu What does "it" refer to in this sentence: " It could be a simple cat from a config file"?

Author: @klgill "it" is the service_command value.

        cat_output: true
        config_mapping:
            ovn-bridge: edpm_ovn_bridge
            ovn-bridge-mappings: edpm_ovn_bridge_mappings
            ovn-encap-type: edpm_ovn_encap_type
            ovn-match-northd-version: ovn_match_northd_version
            ovn-monitor-all: ovn_monitor_all
            ovn-ofctrl-wait-before-clear: edpm_ovn_ofctrl_wait_before_clear
            ovn-remote-probe-interval: edpm_ovn_remote_probe_interval

> ----
> 
> Then you can use this command to compare the values:
> 
> ----
> os-diff diff ovs_external_ids.json edpm.crd --crd --service ovs_external_ids
> ----
> 
> Lets take the example of checking the /etc/yum.conf on every hosts, you will have to put this statement in the config.yaml,
> lets call it yum_config:
> 
> ----
> services:
>   yum_config:
>     hosts:
>       - undercloud
>       - controller_1
>       - compute_1
>       - compute_2
>     service_command: "cat /etc/yum.conf"
>     cat_output: true
>     path:
>       - yum.conf
> ----

Reviewer: ```suggestion
----
+
Then you can use the following command to compare the values:
+
----
os-diff diff ovs_external_ids.json edpm.crd --crd --service ovs_external_ids
----
+
For example, to check the `/etc/yum.conf` on every host, you must put the following statement in the `config.yaml` file. The following example uses a file called `yum_config`:
+
----
services:
  yum_config:
    hosts:
      - undercloud
      - controller_1
      - compute_1
      - compute_2
    service_command: "cat /etc/yum.conf"
    cat_output: true
    path:
      - yum.conf
----
```

Reviewer: @jistr After these changes are committed, we should check how this section renders (proc_pulling-configuration-from-a-tripleo-deployment.adoc). I added several plus signs, and they might not all be necessary, which means they'll appear in the rendered doc.

> 

Reviewer: @matbu I started to format your new section as steps, but some of  this information does not appear to be steps that customers must do to complete the procedure. What information is necessary downstream?

CC: @jistr

Author: @klgill Yes it's not a necessary "step" but more some guide and help for the usage. I put that in place, but if it's sounds confusing we can drop it.

. Pull the configuration:
+

> This command will pull all the configuration files describe into the /etc/os-diff/config.yaml file.

Reviewer 2: "describe into" -> "described in" ?

> This command will pull all the configuration files described in the /etc/os-diff/config.yaml file.
> Os-diff can update this file automaticaly according to your running environment with the command --update or --update-only.
> This option will set the Podman informations into the config.yaml for all running containers.
> It can be useful later, when all the Openstack services will be turned off.
> 
> Note that when the config.yaml is populate automaticaly you have to provide the configuration paths manually for each services.
> 
> ----
> # will only update the /etc/os-diff/config.yaml
> os-diff pull --update-only
> ----

Reviewer: ```suggestion
This command will pull all the configuration files that are described in the `/etc/os-diff/config.yaml` file.
Os-diff can update this file automatically according to your running environment with the command `--update` or `--update-only`.
This option sets the podman information into the `config.yaml` for all running containers.
It can be useful later, when all the {rhos_prev_long} services are turned off.

Note that when the `config.yaml` file is populated automatically you must provide the configuration paths manually for each service.

----
# will only update the /etc/os-diff/config.yaml
os-diff pull --update-only
----
```
